src/clj_jargon/tickets.clj

```diff
@@ -1,33 +1,59 @@
+        [clj-jargon.init :only [anonymous-user-account override-user-account proxy-input-stream]]
         [clj-jargon.cart :only [temp-password]]
         [clj-jargon.item-info :only [file object-type]]
-        [clj-jargon.item-ops :only [input-stream]])
+        [clj-jargon.item-ops :only [input-stream tcb]])
   (:require [clojure-commons.file-utils :as ft]
-            [clojure.tools.logging :as log])
-  (:import [org.irods.jargon.core.pub IRODSAccessObjectFactory]
+            [clojure.tools.logging :as log]
+            [clojure-commons.error-codes :refer [ERR_NOT_WRITEABLE]]
+            [slingshot.slingshot :refer [throw+]])
+  (:import [java.io File]
+           [org.irods.jargon.core.exception CatNoAccessException]
+           [org.irods.jargon.core.pub IRODSAccessObjectFactory]
+           [org.irods.jargon.core.connection IRODSAccount]
            [org.irods.jargon.ticket.packinstr TicketInp] 
            [org.irods.jargon.ticket.packinstr TicketCreateModeEnum] 
            [org.irods.jargon.ticket TicketServiceFactoryImpl
                                     TicketAdminService
                                     TicketAdminServiceImpl
                                     TicketClientSupport
+                                    TicketClientOperations
                                     Ticket]))
 
+(defn- ^IRODSAccount account-for-ticket
+  [cm username]
+  (condp = username
+    (:username cm)
+    (do (log/debug (str "Using existing account since '" username "' = '" (:username cm) "'"))
+        (:irodsAccount cm))
+
+    "anonymous"
+    (anonymous-user-account cm)
+
+    (do (log/debug (str "Creating temporary password for '" username "' which does not match '" (:username cm) "'"))
+        (override-user-account cm username (temp-password cm username)))))
+
 (defn- ^TicketAdminService ticket-admin-service
   "Creates an instance of TicketAdminService, which provides
    access to utility methods for performing operations on tickets.
    Probably doesn't need to be called directly."
   [cm username]
-  (let [tsf (TicketServiceFactoryImpl. (:accessObjectFactory cm))
-        user (if (= username (:username cm))
-                 (do (log/debug (str "Using existing account since '" username "' = '" (:username cm) "'"))
-                     (:irodsAccount cm))
-                 (do (log/debug (str "Creating temporary password for '" username "' which does not match '" (:username cm) "'"))
-                     (override-user-account cm username (temp-password cm username))))]
+  (let [tsf (:ticketServiceFactory cm)
+        user (account-for-ticket cm username)]
     (.instanceTicketAdminService tsf user)))
 
+(defn- ^TicketClientOperations ticket-client-operations
+  "Creates an instance of TicketAdminService, which provides
+   access to utility methods for performing operations on tickets.
+   Probably doesn't need to be called directly."
+  ([cm]
+   (ticket-client-operations cm (:username cm)))
```

Comment:
This shorter version is just for convenience (I may also add it to `ticket-admin-service`)